src/workerd/server/tests/compile-tests/compile-test.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# $1 -> workerd binary path
+# $2 -> path to desired file to compile
+# $3 -> port to curl
+# $4 -> desired output
+
+# Help Function
+function show_help() {
+  echo "
+usage: compile-test.sh [-d] [-h] <workerd command> <file-to-compile> <port-to-curl> <expected-output-file>
+  options:
+    -d print out where tmp files are created and do not delete them
+    -h this help message
+
+  Note: all flags must occur before arguments
+"
+}
+
+DEBUG=false
+
+while getopts "dh" option; do
+  case ${option} in
+    d)
+      DEBUG=true
+      ;;
+    h)
+      show_help
+      exit
+      ;;
+  esac
+done
+
+if $DEBUG; then
+  set -x;
+fi
+
+shift $(expr $OPTIND - 1)
+WORKERD_BINARY=$1
+CAPNP_SOURCE=$2
+PORT=$3
+OK_FILE=$4
+
+COMPILE=$(mktemp)
+$WORKERD_BINARY compile $CAPNP_SOURCE > $COMPILE
+$COMPILE &
+KILL=$!
+OUT=$(mktemp)
+FIXED_OUT=$(mktemp)
+FIXED_OK=$(mktemp)
+sleep 1
```

Comment:
sleep 1 means test will spin for 1 second without doing anything. could we do better here?

Reply:
I think I found a way to do this, please check it.